### PR TITLE
Dev b testbeam

### PR DESCRIPTION
**Pull request by scarletnorberg**

### CODE REVIEW

File: python/modules/BtagSFWeightProducer.py [hunk 1]
```diff
@@ -175,20 +197,32 @@ def analyze(self, event):
...
+                    BTagWeightNLight_up_FS *= 1 - btagSF_up_FS * eff
+                    BTagWeightNLight_down  *= 1 - btagSF_down * eff
+                    BTagWeightNLight_down_FS  *= 1 - btagSF_down_FS * eff
+                    BTagWeightDLight      *= 1 - eff
 
                 BTagWeightD      *= 1 - eff
-
+         
+        if self.FastSim:
+            self.out.fillBranch("BTagWeight_Up_FS",   BTagWeightN_up_FS / BTagWeightD)
```
Comment (zhenbinwu) [inline python/modules/BtagSFWeightProducer.py]: How do we apply this FS uncertainty, if all as a ratio to BTagWeightD?

Comment (pastika) [reply python/modules/BtagSFWeightProducer.py]: you want the ratio of events with "BTagWeight_Up_FS " / "BTagWeight"

The denominator is only a function of the efficiencies, so no SF (fast or fullsim) will affect this.

Comment (zhenbinwu) [reply python/modules/BtagSFWeightProducer.py]: Ha,  BTagWeightD *= eff, but BTagWeightN  *= btagSF * eff *= jet.btagSF*jet.btagSF_F*eff. I got these two mixed up.

**Comment by zhenbinwu** [comment]
The fastsim root files has folders , but that is not in the normal file. Looks like the issue of too many histrograms for root to handle. Has @scarletnorberg tested the fastsim with these root file?

**Comment by zhenbinwu** [comment]
Following up with my last comment. It seems the issue only appear in the GUI. I do get a list of TH2D on command line. However, it is important to test how it run on fastsim signal and the completeness of the signal points.

**Comment by zhenbinwu** [comment]
Ok, Scarlet showed it work. So merging it now

**Pull request merged**